### PR TITLE
feat(ofx): add ofx importer

### PR DESCRIPTION
**Pull request by sevazhidkov**

### CODE REVIEW

File: imports/ofx/account.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+				Transactions []Transaction `xml:"STMTTRN"`
+			} `xml:"BANKTRANLIST"`
+		} `xml:"STMTRS"`
+	}
+
+	if err := decoder.DecodeElement(&data, &start); err != nil {
+		return err
+	}
+
+	*a = Account(data.Alias)
```
Comment (shekhirin) [inline imports/ofx/account.go:43]: as I see, we don't parse any fields into the `Alias` struct we embed and use only those specified manually in the inlined struct. Do we need an `Alias` here?
```suggestion
	var data struct {
		AccountInfo struct {
			Currency string `xml:"CURDEF"`
			Meta     struct {
				BankID    string `xml:"BANKID"`
				AccountID string `xml:"ACCTID"`
				Type      string `xml:"ACCTTYPE"`
			} `xml:"BANKACCTFROM"`
			TransactionsStruct struct {
				Transactions []Transaction `xml:"STMTTRN"`
			} `xml:"BANKTRANLIST"`
		} `xml:"STMTRS"`
	}

	if err := decoder.DecodeElement(&data, &start); err != nil {
		return err
	}
```

File: imports/ofx/account.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	var data struct {
+		Alias
+		AccountInfo struct {
+			Currency string `xml:"CURDEF"`
+			Meta     struct {
+				BankID    string `xml:"BANKID"`
+				AccountID string `xml:"ACCTID"`
+				Type      string `xml:"ACCTTYPE"`
+			} `xml:"BANKACCTFROM"`
+			TransactionsStruct struct {
```
Comment (shekhirin) [inline imports/ofx/account.go:33]: `TransactionsStruct` name seems too code-related for me 😬  How's about something like this more related to the origin XML?
```suggestion
			BankTransactionsList struct {
```

File: imports/ofx/account.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	if err := decoder.DecodeElement(&data, &start); err != nil {
+		return err
+	}
+
+	*a = Account(data.Alias)
+	a.Currency = data.AccountInfo.Currency
+	a.BankID = data.AccountInfo.Meta.BankID
+	a.AccountID = data.AccountInfo.Meta.AccountID
+	a.Type = data.AccountInfo.Meta.Type
+	a.Transactions = append(a.Transactions, data.AccountInfo.TransactionsStruct.Transactions...)
```
Comment (shekhirin) [inline imports/ofx/account.go:48]: I see that all other fields are reassigned and only this one changed with respect to the previous state. Is that ok?
```suggestion
	a.Transactions = data.AccountInfo.TransactionsStruct.Transactions
```

File: imports/ofx/datetime.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	"database/sql/driver"
+	"fmt"
+	"github.com/aclindsa/ofxgo"
+	"github.com/aclindsa/xml"
+	"github.com/mattn/go-sqlite3"
+	"time"
+)
+
+// 20210310002816.000[+3:MSK]
+// 20210318
```
Comment (shekhirin) [inline imports/ofx/datetime.go:13]: Don't quite get what these comments mean 😬

File: imports/ofx/statement_test.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+				CurrencyRate: 0,
+			},
+		},
+	}
+	assertAccount(t, expectedAccount, accounts[0])
+
+	var newAccounts []Account
+	require.NoError(t, p.DB().Preload("Transactions").Find(&newAccounts).Error)
+	require.Len(t, newAccounts, 1)
+	assertAccount(t, accounts[0], newAccounts[0])
```
Comment (shekhirin) [inline imports/ofx/statement_test.go:50]: What happens here? Looks like you're querying same DB without any changes once again

File: imports/ofx/datetime.go (hunk 4 above)
Comment (shekhirin) [reply imports/ofx/datetime.go:13]: Ok, I got it. It's two timestamp formats from the OFX file. How's about getting rid of `ofxgo` lib just for parsing datetimes and do it by ourselves? Seems reasonable for me to not include the whole library just for the datetime parsing.

Comment (sevazhidkov) [reply imports/ofx/datetime.go:13]: Yup, my bad, forgot to delete the comments.

I thought of getting rid of `ofxgo`, but decided that copy-pasting this function is worse than an additional dependency: https://github.com/aclindsa/ofxgo/blob/3240ef383b1eea732315a3034c78e69cf4f2d886/types.go#L106

File: imports/ofx/account.go (hunk 3 above)
Comment (sevazhidkov) [reply imports/ofx/account.go:48]: nope, my mistake.

File: imports/ofx/statement_test.go (hunk 5 above)
Comment (sevazhidkov) [reply imports/ofx/statement_test.go:52]: yup, forgot to add re-import.

File: imports/ofx/account.go [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+		AccountInfo struct {
+			Currency string `xml:"CURDEF"`
+			Meta     struct {
+				BankID    string `xml:"BANKID"`
+				AccountID string `xml:"ACCTID"`
+				Type      string `xml:"ACCTTYPE"`
+			} `xml:"BANKACCTFROM"`
+			TransactionsInfo struct {
+				Transactions []Transaction `xml:"STMTTRN"`
+			} `xml:"BANKTRANLIST"`
```
Comment (shekhirin) [inline imports/ofx/account.go:32]: > * If the XML element contains a sub-element whose name matches
   the prefix of a tag formatted as "a" or "a>b>c", unmarshal
   will descend into the XML structure looking for elements with the
   given names, and will map the innermost elements to that struct
   field. A tag starting with ">" is equivalent to one starting
   with the field name followed by ">".

from https://golang.org/pkg/encoding/xml/#Unmarshal

---

I think we can do similar thing here
```suggestion
			Transactions []Transaction `xml:"BANKTRANLIST>STMTTRN"`
```

Don't know if it works, just a suggestion 😄

**Review by shekhirin — APPROVED** [review]

**Pull request merged**